logic/src/commonTest/kotlin/com/wire/kalium/logic/data/call/CallRepositoryTest.kt

```diff
@@ -41,7 +45,175 @@ class CallRepositoryTest {
         }
     }
 
+    @Test
+    fun givenEmptyListOfCalls_whenGetAllCallsIsCalled_thenReturnAnEmptyListOfCalls() {
+        val calls = callRepository.getAllCalls()
+
+        assertEquals(0, calls.value.size)
+    }
+
+
+    @Test
+    fun givenAListOfCallProfiles_whenGetAllCallsIsCalled_thenReturnAnEmptyListOfCalls() {
+        callRepository.updateCallProfileFlow(
+            CallProfile(
+                mapOf(
+                    startedCall.conversationId.toString() to startedCall,
+                    answeredCall.conversationId.toString() to answeredCall,
+                    incomingCall.conversationId.toString() to incomingCall,
+                    establishedCall.conversationId.toString() to establishedCall
+                )
+            )
+        )
+
+        val calls = callRepository.getAllCalls()
+
+        assertEquals(4, calls.value.size)
+        assertEquals(startedCall, calls.value[0])
+        assertEquals(establishedCall, calls.value[3])
+    }
+
+
+    @Test
+    fun givenACallObject_whenCreateCallCalled_thenAddThatCallToTheFlow() {
+        callRepository.updateCallProfileFlow(CallProfile(mapOf(startedCall.conversationId.toString() to startedCall)))
+
+        callRepository.createCall(answeredCall)
+
+        val calls = callRepository.getAllCalls()
+        assertEquals(2, calls.value.size)
+        assertEquals(calls.value[0], startedCall)
+        assertEquals(calls.value[1], answeredCall)
+    }
+
+    @Test
+    fun givenCallObjectWithSameConversationIdAsAnotherInTheFlow_whenCreateCallCalled_thenEraseTheOldOneWithTheNewer() {
```

Comment:
Instead of "erase the old one with the newer", what about "update the current" or "replace the current"?